tests/unit/test_coloring.py

```diff
@@ -0,0 +1,200 @@
+        # Adds new vertices
+        self.coloring.add([self.v1,self.v2], 4)
+        vertices = self.coloring.get(4)
+        self.assertEqual(2, len(vertices))
+        self.assertEqual(True, self.v1 in vertices)
+        self.assertEqual(True, self.v2 in vertices)
+
+        # Recolors old vertices
+        v3, v4 = Vertex(self.graph), Vertex(self.graph)
+        self.coloring.set(v3, 4)
+        self.coloring.add([self.v1, self.v2, v4], 0)
+        vertices = self.coloring.get(0)
+        self.assertEqual(3, len(vertices))
+        self.assertEqual(True, self.v1 in vertices)
+        self.assertEqual(True, self.v2 in vertices)
+        self.assertEqual(True, v4 in vertices)
+        self.assertEqual(False, v3 in vertices)
+        self.assertEqual(1, len(self.coloring.get(4)))
+
+    def test_color(self):
+        self.coloring.set(self.v1, 3)
+        self.coloring.set(self.v2, 4)
+        self.assertEqual(3, self.coloring.color(self.v1))
+        self.assertEqual(4, self.coloring.color(self.v2))
+
+    def test_recolor(self):
+        self.coloring.set(self.v1, 1)
+        self.assertEqual(1, self.coloring.color(self.v1))
+
+        self.coloring.recolor(self.v1, 2)
+        self.assertEqual(2, len(self.coloring))
+        self.assertEqual(False, self.v1 in self.coloring.get(1))
+        self.assertEqual(True, self.v1 in self.coloring.get(2))
+
+        # Cannot recolor vertex that is not in the coloring
+        with self.assertRaises(KeyError) as e:
+            self.coloring.recolor(self.v2, 2)
+
+        self.assertEqual("\'Vertex {} not found in coloring\'".format(str(self.v2)), str(e.exception))
+
+    def test_colors(self):
+        self.coloring.set(self.v1, 0)
+        self.coloring.set(self.v2, 1)
+        self.coloring.set(Vertex(self.graph), 2)
+
+        self.assertEqual(3, len(self.coloring))
+        self.assertIn(0, self.coloring.colors)
+        self.assertIn(1, self.coloring.colors)
+        self.assertIn(2, self.coloring.colors)
+
+    def test_vertices(self):
+        v3 = Vertex(self.graph)
+        self.coloring.set(self.v1, 0)
+        self.coloring.set(self.v2, 1)
+        self.coloring.set(v3, 1)
+
+        self.assertEqual(3, len(self.coloring.vertices))
+        self.assertIn(self.v1, self.coloring.vertices)
+        self.assertIn(self.v2, self.coloring.vertices)
+        self.assertIn(v3, self.coloring.vertices)
+
+    def test_next_color(self):
+        for i in range(10):
+            self.assertEqual(i, len(self.coloring))
+            self.assertEqual(i, self.coloring.next_color())
+            self.coloring.set(Vertex(self.graph), i)
+
+    def test_items(self):
+        self.coloring.set(self.v1, 0)
+        self.coloring.set(self.v2, 1)
+        self.coloring.set(Vertex(self.graph), 2)
+
+        for c, v in self.coloring.items():
+            self.assertIn(c, self.coloring.colors)
+            self.assertEqual(self.coloring.get(c),list(v))
+
+    def test_status(self):
+        # For different graphs
```

Comment:
Zou je een plaatje van iedere graaf willen toevoegen? Dan is het makkelijk om te zien met wat voor graaf je test